src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/UserManagement/UsersController.cs

```diff
@@ -28,77 +24,56 @@ public UsersController(IUserManagementService userManagementService)
 
         [HttpGet("user-management/users")]
         [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
         public async Task<ActionResult<List<UserViewModel>>> GetUserList()
         {
-            var users = await _userManagementService.ListAsync();
-
-            if (users.Any())
-            {
-                return Ok(users);
-            }
-
-            return NotFound();
+            return await _userManagementService
+                .ListAllUsers()
+                .HandleFailuresOrOk();
         }
 
         [HttpGet("user-management/users/{userId}")]
         [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
         public async Task<ActionResult<UserViewModel>> GetUser(string userId)
         {
-            var user = await _userManagementService.GetAsync(userId);
-
-            if (user != null)
-            {
-                return Ok(user);
-            }
-
-            return NotFound();
+            return await _userManagementService
+                .GetUser(userId)
+                .HandleFailuresOrOk();
         }
 
         [HttpPut("user-management/users/{userId}")]
-        public async Task<ActionResult<UserViewModel>> UpdateUser(string userId, EditUserViewModel model)
+        public async Task<ActionResult<Unit>> UpdateUser(string userId, EditUserViewModel model)
         {
-            var user = await _userManagementService.UpdateAsync(userId, model.RoleId);
-
-            if (user)
-            {
-                return Ok(true);
-            }
-
-            AddErrors(ModelState, ValidationResult(UserDoesNotExist));
-
-            return ValidationProblem(new ValidationProblemDetails(ModelState));
+            return await _userManagementService
+                .UpdateUser(userId, model.RoleId)
+                .HandleFailuresOrOk();
         }
 
         [HttpGet("user-management/users/{userId}/release-role")]
         [ProducesResponseType(200)]
-        [ProducesResponseType(404)]
-        public ActionResult<List<UserReleaseRoleViewModel>> GetUserReleaseRoles(Guid userId)
+        public Task<ActionResult<List<UserReleaseRoleViewModel>>> GetUserReleaseRoles(string userId)
```

Comment:
I think with a little front end refactoring this could be scrapped as the roles are already fetched when it fetches the user.

Reply:
Created a separate ticket to address this: https://dfedigital.atlassian.net/browse/EES-1787